fancy_prompt/fancy_prompt.py

```diff
@@ -0,0 +1,189 @@
+'''
+Enhanced prompt support for UndoDB.
+Contributors: Mark Williamson, Toby Lloyd Davies
+Copyright (C) 2019 Undo Ltd
+'''
+
+from __future__ import absolute_import, division
+import re
+import datetime
+from undodb.debugger_extensions import udb
+import gdb
+
+
+# Configuration variable: apply colour to output?  Disable for TUI mode.
+colour_enabled = False
+
+
+class PromptColour(gdb.Command):
+    '''
+    Set color on or off for enhanced UndoDB prompt.
+
+    Usage: prompt-color SETTING
+
+    Arguments:
+        SETTING - on|off to enable / disable colors.
+    '''
+
+    def __init__(self):
+        super(PromptColour, self).__init__('prompt-color', gdb.COMMAND_SUPPORT)
+
+    @staticmethod
+    def invoke(argument, from_tty):
+        global colour_enabled
+        if argument == 'on':
+            colour_enabled = True
+        elif argument == 'off':
+            colour_enabled = False
+
+
+PromptColour()
+
+
+def undodb_get_extent():
+    '''
+    Get the event log extent of the current debuggee.
+
+    Returns: (bb_start, bb_end) tuple of start and end bbcounts.
+    '''
+    undodb_state = undodb_get_state()
+    if undodb_state == STATE_DEFER or undodb_state == STATE_NONE:
+        return None
+    return udb.get_event_log_extent()
```

Comment:
What happens if you don't do the check first?

Reply:
state_none: udbpy.comms.NotRunningError
state_defer: udbpy.comms.NotConnectedError